src/main/java/org/nuxeo/binary/metadata/internals/BinaryMetadataServiceImpl.java

```diff
@@ -197,16 +198,25 @@ public void writeMetadata(DocumentModel doc, String mappingDescriptorId) {
             // Write doc properties from outputs.
             for (String metadata : blobMetadataOutput.keySet()) {
                 Object metadataValue = blobMetadataOutput.get(metadata);
+                boolean metadataIsArray = metadataValue.getClass().isArray() || metadataValue instanceof ArrayList;
```

Comment:
`metadataValue.getClass().isArray()` —> `metadataValue instanceof Object[]` is more readable (and `null`-proof).